### ISSUE TITLE
Edit PDF text directly

### ISSUE THREAD

**Issue by alwin0516** [Stirling-Tools/Stirling-PDF#1141]
I found this [project](https://github.com/FranBar1966/ncSimpleHtmlEditor) (https://github.com/FranBar1966/ncSimpleHtmlEditor), Can this be used to edit PDF text?

Edit the text on the PDF directly through the UI.

may need to convert PDF to HTML, and then convert to PDF after editing the text.

**Comment by Barikad** [comment]
I agree, the possibility of editing the text of a PDF, while retaining the fonts (standards or embedded in the PDF) would be a real plus and would make it possible to completely do without online services that "offer" this functionality

**Comment by cybergrunge** [comment]
This is indeed the last feature missing compared to closed-source solutions.

**Comment by johnvogelbangzam** [comment]
Tambem gostaria muito dessa funcionalidade

**Comment by r4lix** [comment]
Would love this, any update ?

**Comment by renatoaraujoc** [comment]
This is the absolute missing functionality to make us ditch Adobe Pro here.
Any idea when it would be implemented?

**Comment by cracksilver** [comment]
me to. This would be the infinity feature.

**Comment by cracksilver** [comment]
This was the main goal I wanted to go with Stirling-PDF. But I think there is now update for this function.

**Comment by Frooodle** [comment]
Currently still not direct text editting 
We can add text fine but not edit existing text

**Comment by y0d4a** [comment]
it is not easy to implement some existing solutions? like old pdf-lib? https://github.com/Hopding/pdf-lib

**Comment by Frooodle** [comment]
We already use pdf lib for some features actually 
But also it doesn't support text editting

**Comment by Elektroniker0486** [comment]
Yes, if the “edit text” function is implemented, then your tool will be perfect! I hope for a prompt implementation. Many thanks in advance

**Comment by twinmoonsllc** [comment]
99% of editing I do to a PDF is changing the text of the PDF. This is the most critical feature! Other than that, this is such a great project. Alas, I still can't ditch Acrobat Pro.

**Comment by yukikawas** [comment]
This is a feature that many people are looking forward to, and if implemented, it should bring in more users.

**Comment by gofort** [comment]
Installed this software just to edit text in some PDFs I need and this is missing lol what a bad luck, I thought this is a must have for any pdf editor
hope it will be added soon

**Comment by scottwegner** [comment]
Agreed!  That would be an amazing feature.

**Comment by karimz1** [comment]
Yep I agree too

**Comment by davidhantson** [comment]
Hey, 

longtime user of ilovepdf.com cant remember if i ever edited text live in a pdf, adding and filling in text, yes ... editing is just convert to word and save again as pdf?

is it possible with Stirling?

**Comment by varunkotian** [comment]
This is a much-needed feature. 
Only feature holding this back to disrupt Adobe

**Comment by Lindismash1** [comment]
This is a feature i think everyone using Stirling-PDF is waiting for..
I wonder whats the progress about this.

**Comment by tomokp** [comment]
Is this feature planned, or is this not something that will ever be added?
If it is planned, what's the expected timeline on this?
Only missing feature for me, and a lot of other people it seems, to ditch all other pdf editors.

**Comment by Frooodle** [comment]
We are working on Stirling-PDF V2 at the moment, which will add a ton of improvements and features and we hope will be the backbone towards greater features like edit text.
But I cant give any estimates on the edit text feature itself, its a large unknown and we have other things in our backlog along with it
BUT yes it is planned

**Comment by tomokp** [comment]
> We are working on Stirling-PDF V2 at the moment, which will add a ton of improvements and features and we hope will be the backbone towards greater features like edit text. But I cant give any estimates on the edit text feature itself, its a large unknown and we have other things in our backlog along with it BUT yes it is planned

Appreciate the engagement with the community. Looking forward to V2!

**Comment by 0gawan** [comment]
> We are working on Stirling-PDF V2 at the moment, which will add a ton of improvements and features and we hope will be the backbone towards greater features like edit text. But I cant give any estimates on the edit text feature itself, its a large unknown and we have other things in our backlog along with it BUT yes it is planned

It's planned For Sure ?
This Most attended Features, seems to be "On Hold/Dropped" in V1 (https://github.com/orgs/Stirling-Tools/projects/4/views/1?filterQuery=edit)
And i don't see anythings in the Features List of [V2](https://github.com/orgs/Stirling-Tools/projects/12/views/1)  , some missings ?
This is a great tool, your work is awesome, keep going !!!!

**Comment by Frooodle** [comment]
We now have an initial extremely early alpha version releeased fully Opensource.. One ask for the community is for PDFs that dont render correctly at all (on export) can you please share the font info shown in bottom right of page.
